compiler/shared/main/scala/mlscript/compiler/ir/Builder.scala

```diff
@@ -243,7 +245,9 @@ final class Builder(fresh: Fresh, fnUid: FreshInt, classUid: FreshInt, tag: Fres
     res
   
   private def buildDefFromNuFunDef(using ctx: Ctx)(nfd: Statement): Defn = nfd match
-    case NuFunDef(_, Var(name), None, Nil, L(Lam(Tup(fields), body))) =>
+    case nfd: NuFunDef =>
+      val NuFunDef(_, Var(name), None, Nil, L(Lam(Tup(fields), body))) = nfd
```

Comment:
Note that this partial destructuring is unsafe. Conceptually, the lambda `(...xs) => ...` will not have a tuple on its LHS. Though that syntax is currently not supported.